core/src/main/java/brooklyn/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -322,45 +309,19 @@ public void load() {
         }
     }
 
-    @SuppressWarnings("unchecked")
-    @Override
     public <T, SpecT> SpecT createSpec(CatalogItem<T, SpecT> item) {
```

Comment:
This is public API so I'd be way of changing it, plus `catalog` is already implicit because it's a method on the catalog interface.